test/suites/upgrade/pre-upgrade-setup.yaml

```diff
@@ -0,0 +1,124 @@
+      script: |
+        TOKEN=$(curl -X PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 21600")
+        curl -H "X-aws-ec2-metadata-token: $TOKEN" -v http://169.254.169.254/latest/meta-data/identity-credentials/ec2/info | jq -r ".AccountId" | tr -d '\n' > $(results.account-id.path)
+        curl -H "X-aws-ec2-metadata-token: $TOKEN" -v http://169.254.169.254/latest/meta-data/placement/region > $(results.region.path)
+
+    - name: checkout-repository
+      image: alpine/git
+      workingDir: $(workspaces.ws.path)
+      script: |
+        echo "Cloning $(params.git-repo-url)"
+        echo "Checking out git ref: $(params.git-ref)"
+        git clone $(params.git-repo-url).git .
+        git checkout $(params.git-ref)
+
+    - name: create-iam-policies
+      image: public.ecr.aws/karpenter-testing/tools:latest
+      workingDir: $(workspaces.ws.path)
+      script: |
+        aws iam create-service-linked-role --aws-service-name spot.amazonaws.com || true
+        aws cloudformation deploy \
+          --stack-name KarpenterIAM-$(params.cluster-name) \
+          --template-file website/content/en/preview/getting-started/getting-started-with-eksctl/cloudformation.yaml \
+          --capabilities CAPABILITY_NAMED_IAM \
+          --parameter-overrides "ClusterName=$(params.cluster-name)"
+
+    - name: create-cluster
+      image: public.ecr.aws/karpenter-testing/tools:latest
+      script: |
+        cmd="create"
+        eksctl get cluster --name $(params.cluster-name) && cmd="upgrade"
+        eksctl ${cmd} cluster -f - <<EOF
+        ---
+        apiVersion: eksctl.io/v1alpha5
+        kind: ClusterConfig
+        metadata:
+          name: $(params.cluster-name)
+          region: $(cat $(results.region.path))
+          version: "$(params.kubernetes-version)"
+          tags:
+            karpenter.sh/discovery: $(params.cluster-name)
+        kubernetesNetworkConfig:
+          ipFamily: $(params.ip-family)
+        managedNodeGroups:
+          - instanceType: m5.large
+            amiFamily: AmazonLinux2
+            name: $(params.cluster-name)-system-pool
+            desiredCapacity: 2
+            minSize: 2
+            maxSize: 2
+            taints:
+            - key: CriticalAddonsOnly
+              value: "true"
+              effect: NoSchedule
+        iam:
+          withOIDC: true
+        addons:
+        - name: vpc-cni
+        - name: coredns
+        - name: kube-proxy
+        - name: aws-ebs-csi-driver
+          wellKnownPolicies: 
+            ebsCSIController: true
+        EOF
+        
+        eksctl create iamidentitymapping \
+        --username system:node:{{EC2PrivateDNSName}} \
+        --cluster "$(params.cluster-name)" \
+        --arn "arn:aws:iam::$(cat $(results.account-id.path)):role/KarpenterNodeRole-$(params.cluster-name)" \
+        --group system:bootstrappers \
+        --group system:nodes
+        
+        eksctl create iamserviceaccount \
```

Comment:
Note that the installation and the policy ARNs are different from the pure `setup.yaml`